src/interchange/src/avro/decode.rs

```diff
@@ -438,6 +440,48 @@ impl<'a> AvroDecode for AvroFlatDecoder<'a> {
         ));
         Ok(())
     }
+
+    #[inline]
+    fn numeric<'b, R: AvroRead>(
+        self,
+        _precision: usize,
+        scale: usize,
+        r: ValueOrReader<'b, &'b [u8], R>,
+    ) -> Result<Self::Out, AvroError> {
+        let buf = match r {
+            ValueOrReader::Value(val) => val,
+            ValueOrReader::Reader { len, r } => {
+                self.buf.resize_with(len, Default::default);
+                r.read_exact(self.buf)?;
+                &self.buf
+            }
+        };
+        let coefficient =
+            rdn::twos_complement_be_to_i128(buf).map_err(|e| DecodeError::Custom(e.to_string()))?;
+        let mut cx = DecimalCx::<Decimal128>::default();
+        let mut n = cx.from_i128(coefficient);
+        cx.set_exponent(&mut n, -(scale as i32));
+
+        let n = OrderedDecimal(n);
+
+        if rdn::check_max_precision_strict(&cx, &n).is_err() {
```

Comment:
What do we intend to happen if someone has an Avro schema with a decimal field where `precision` is greater than 16?

Reply:
The max precision is 34, but the idea here was just to treat it as if it were ~= an overflow and disallow it from being input. lmk if that makes sense!

Reply:
Well, it's a regression - it will mean we can't ingest avro sources with schemas that we currently can. Sorry, I'm not up to speed on the discussions about decimal types so perhaps the issues the new type is fixing is important enough that we're OK with just telling people with precision > 34 "sorry, you can't use Materialize."

Anyway, if we _do_ have a hard requirement of max precision 34, we should probably make sure to catch it at source creation time when we parse schemas, in addition to catching it here.

Reply:
> The max precision is 34

Is this not a regression from the current design?

Reply:
Just for those following along at home, this is a regression, but the current implementation's guarantee of 38 digits of precision doesn't actually hold up in some cases, e.g.

```
SELECT 1::NUMERIC(38, 10) / 1.2103500000::numeric(38, 10);
----
0.826000000000
```

The correct answer to this query should be ~`0.82620729541`, so we're offering 38 digits of precision, but then ~losing 7 at the end there, whereas this new library is capable of producing the correct result.